ckanext/datastore/tests/test_idatadictionaryform.py

```diff
@@ -0,0 +1,153 @@
+# encoding: utf-8
+
+import pytest
+import ckan.plugins as p
+import ckan.tests.helpers as helpers
+import ckan.tests.factories as factories
+from ckan.plugins.toolkit import ValidationError
+from ckanext.datastore.backend import DatastoreBackend
+
+@pytest.mark.ckan_config(
+    "ckan.plugins", "datastore example_idatadictionaryform")
+@pytest.mark.usefixtures("clean_datastore", "with_plugins")
+def test_accept_custom_fields():
+    fields = [
+        {
+            "id": "one", "type": "text",
+            "an_int": 42, "json_obj": {"a": "b"}, "secret": 7,
+        },
+        {
+            "id": "two", "type": "text",
+            "an_int": -4, "json_obj": {"q": []},
+        },
+    ]
+    resource = _create_datastore_resource(fields)
+
+    result = helpers.call_action(
+        "datastore_info", id=resource["id"]
+    )
+
+    f1 = {k: v for k, v in result['fields'][0].items() if k != 'schema'}
+    f2 = {k: v for k, v in result['fields'][1].items() if k != 'schema'}
+    assert f1 == {
+        "id": "one", "type": "text", "an_int": 42, "json_obj": {"a": "b"}}
+    assert f2 == {
+        "id": "two", "type": "text", "an_int": -4, "json_obj": {"q": []}}
+
+    backend = DatastoreBackend.get_active_backend()
+
+    plugin_data = backend.resource_plugin_data(resource["id"])
+
+    assert plugin_data['one']['example_idatadictionaryform_secret'] == {
+        'secret': 7}
+    assert 'example_idatadictionaryform_secret' not in plugin_data['two']
+
+    # excluding/passing empty values for all fields for a plugin data
+    # key will leave existing values unchanged due to ignore_empty
```

Comment:
@amercader this is an interaction between the plugin_data implementation (no old data is sent to validators) and the way ignore_empty works. There's no way to clear all existing values if the plugin uses ignore_empty as a validator. WDYT? Should we special case passing empty but not missing values so they force old values to be cleared somehow (a more complicated validator?)

Reply:
@KatiRG IDataDictionaryForm would be a great fit for your validation-rules-in-the-data-dictionary feature. How would you like to have deleting data dictionary values work from an API perspective?